src/utilitis/detectionsConverter.ts

```diff
@@ -0,0 +1,75 @@
+import DetectorOpts, { DetectionConvertion } from './../interfaces/IDetectorOpts';
+import Logger from './logger';
+
+const _filterDetections = (objectFilter: Array<String>, detections: Array<any>): Array<any> =>
+  objectFilter.length === 0
+    ? detections
+    : detections.filter(({ label }) => objectFilter.some(x => x === label));
+
+export default class DetectionsConverter {
+  _detections: Array<any>;
+  _framingBBox: { x: number; y: number; height: number; width: number };
+  _detectionConvertion: DetectionConvertion;
+  _preview_image_size: { height: number; width: number };
+  _objectFilter: Array<String> = ['head', 'person'];
+
+  constructor(
+    detections: Array<any>,
+    preview_image_size: { height: number; width: number },
+    options: DetectorOpts,
+    frame?: { bbox: { x: number; y: number; height: number; width: number } }
+  ) {
+    this._detections = detections;
+    this._preview_image_size = preview_image_size;
+    this._validate(options, frame);
+    if (options && options.objectFilter) {
+      this._objectFilter = options.objectFilter;
+    }
+    this._detectionConvertion = options.convertDetections;
+    if (frame && frame.bbox) {
+      this._framingBBox = frame.bbox;
+    }
+  }
+
+  _validate(
+    options: DetectorOpts,
+    frame: { bbox: { x: number; y: number; height: number; width: number } }
+  ): void {
+    if (options && options.convertDetections === DetectionConvertion.FRAMING && !frame) {
+      throw new Error('Cannot convert to FRAMING coordinates without a provided frame');
+    }
+  }
+
+  convert(): Array<any> {
```

Comment:
We should probably add a unit test for this.

Reply:
convertion tests have been moved to detectionsConverter.spec.ts